scripts/task.ts

```diff
@@ -315,7 +316,13 @@ async function runTask(task: Task, details: TemplateDetails, optionValues: Passe
     const hasJunitFile = await pathExists(junitFilename);
     // If there's a non-test related error (junit report has not been reported already), we report the general failure in a junit report
     if (junitFilename && !hasJunitFile) {
-      await writeJunitXml(getTaskKey(task), details.key, startTime, err, true);
+      await writeJunitXml(
+        getTaskKey(task),
+        details.key,
+        startTime,
+        err instanceof Error ? err : new Error(String(err)),
```

Comment:
I'm not sure what `err` might be here so this seemed safest. But if it's expected always to be an `Error` it might be better to make `writeJunitXml` take `unknown` and use an invariant similar to what `sendTelemetryError` does in #23632 .